src/averageAges.js

```diff
@@ -55,7 +69,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothersWithChild = people.filter(person => {
+    const mother = people.find(m => m.name === person.mother);
+
+    return onlyWithSon
+      ? mother && person.sex === 'm'
+      : mother;
+  });
+
+  const totalAgeDiff = mothersWithChild.reduce((sum, person) => {
+    const mother = people.find(m => m.name === person.mother);
+
+    return sum + (person.born - mother.born);
+  }, 0);
+
+  const averageAgeDiff = totalAgeDiff / mothersWithChild.length || 0;
```

Comment:
Create separate function `calculateAverageAge` and move this logic to this function.
Now you have the logic with reduce repeated in all 3 functions.
Make it so that the function takes an array of numbers and the length by which it needs to be divided